src/k2/dto/Expression.h

```diff
@@ -0,0 +1,106 @@
+#include <k2/transport/Payload.h>
+#include <k2/common/Common.h>
+#include <k2/transport/PayloadSerialization.h>
+#include <vector>
+#include <functional>
+#include "FieldTypes.h"
+#include "SKVRecord.h"
+
+namespace k2 {
+namespace dto {
+namespace expression {
+
+struct InvalidExpressionException : public std::exception {
+    virtual const char* what() const noexcept override{ return "InvalidExpression";}
+};
+
+struct TypeMismatchException : public std::exception {
+    virtual const char* what() const noexcept override { return "TypeMismatch"; }
+};
+
+enum class Operation : uint8_t {
+    EQ,           // value operands only
+    GT,           // value operands only
+    GTE,          // value operands only
+    LT,           // value operands only
+    LTE,          // value operands only
+    IS_NULL,      // value operands only
+    IS_TYPE,      // value operands only
+    STARTS_WITH,  // value operands only
+    CONTAINS,     // value operands only
+    ENDS_WITH,    // value operands only
+    AND,          // mixed value and/or nested expression operands
+    OR,           // mixed value and/or nested expression operands
+    XOR,          // mixed value and/or nested expression operands
+    NOT,           // mixed value and/or nested expression operands
+    UNKNOWN
+};
+
+// A Value in the expression model. It can be either
+// - a field reference which sets the fieldName, or
```

Comment:
Should we consider using schema Id + field Idx instead of fieldName? Since when the client construct the col/field reference, it has already resolved the column/field name with schema, so using fieldName still will be less efficient as it need to be resolved again, as in SchematizedValue constructor?